index.js

```diff
@@ -1,10 +1,18 @@
 const core = require('@actions/core');
 const exec = require('@actions/exec');
 const github = require('@actions/github');
-
-const ARGS = [
-    'run', '--all-files', '--show-diff-on-failure', '--color=always'
-];
+const tr = require('@actions/exec/lib/toolrunner');
+
+let args = core.getInput('args');
+if (!args) {
+    args = [
+        'run', '--all-files', '--show-diff-on-failure', '--color=always'
+    ];
+} else {
+    args = [
+        'run', ...tr.argStringToArray(args), '--show-diff-on-failure', '--color=always'
+    ];
+}
```

Comment:
- this shouldn't be done at the module level (~generally you want to avoid side-effects at the global scope), let's put it inside `main` -- this also means any generic error handling done in `main` will also apply to this logic
- you'll need to add this key to `action.yml` so it is advertised
- we'll need some documentation of this in the README
- let's call it `extra_args`
- let's require it to be an array so we don't need to parse it
- if it defaults to an empty array you can just splat it on the end instead of the middle (argument order doesn't matter with the argument parsing pre-commit provides)

Reply:
Thanks for the feedback! I'll take a stab at addressing these tonight.

Reply:
@asottile The above are all resolved! Happy to revise as necessary.

I defaulted `extra_args` to `['--all-files']`, aligned with current behavior.